examples/md-flexible/src/configuration/YamlParser.cpp

```diff
@@ -503,6 +496,91 @@ bool MDFlexParser::YamlParser::parseYamlFile(MDFlexConfig &config) {
+        };
+
+        for (auto siteIterator = node[MDFlexConfig::siteStr].begin(); siteIterator != node[MDFlexConfig::siteStr].end();
+             ++siteIterator) {
+          siteErrors.clear();
+          siteID = std::distance(node[MDFlexConfig::siteStr].begin(), siteIterator);
+
+          const auto epsilon =
+              parseComplexTypeValueSingle<double>(siteIterator->second, config.epsilonMap.name.c_str(), siteErrors);
+          const auto sigma =
+              parseComplexTypeValueSingle<double>(siteIterator->second, config.sigmaMap.name.c_str(), siteErrors);
+          const auto mass =
+              parseComplexTypeValueSingle<double>(siteIterator->second, config.massMap.name.c_str(), siteErrors);
+
+          config.addSiteType(siteID, epsilon, sigma, mass);
+        }
+      } else if (key == MDFlexConfig::moleculesStr) {
+        // todo throw error if momentOfInertia with zero element is used (physically nonsense + breaks the quaternion
+        // update)
+        expected = "See AllOptions.yaml for examples.";
+        description = "";
+
+        // remove default objects
+        config.molToSiteIdMap.clear();
+        config.molToSitePosMap.clear();
+        config.momentOfInertiaMap.clear();
+
+        int molID = 0;
+        std::vector<std::string> molErrors;
+
+        auto pushMolError = [&](const std::string &error) {
+          std::stringstream ss;
+          ss << "YamlParser: Error parsing multi-site molecule with ID " << molID << "." << std::endl
+             << "Message: " << error << std::endl
+             << "See AllOptions.yaml for examples." << std::endl;
+          errors.push_back(ss.str());
+        };
+
+#if MD_FLEXIBLE_MODE == MULTISITE
+        for (auto molIterator = node[MDFlexConfig::moleculesStr].begin();
+             molIterator != node[MDFlexConfig::moleculesStr].end(); ++molIterator) {
+          molErrors.clear();
+          molID = std::distance(node[MDFlexConfig::moleculesStr].begin(), molIterator);
+
+          const auto molToSiteId = parseComplexTypeValueSequence<unsigned long>(
+              molIterator->second, MDFlexConfig::moleculeToSiteIdStr, molErrors);
+          const auto molToSitePos = parseComplexTypeValueSequence<std::array<double, 3>>(
+              molIterator->second, MDFlexConfig::moleculeToSitePosStr, molErrors);
+          const auto momentOfInertia = parseComplexTypeValueSequence<double, 3>(
+              molIterator->second, MDFlexConfig::momentOfInertiaStr, molErrors);
+
+          if (molToSiteId.size() != molToSitePos.size()) {
+            std::stringstream ss;
+            ss << "Lengths of " << MDFlexConfig::moleculeToSiteIdStr << " and " << MDFlexConfig::moleculeToSitePosStr
+               << "do not match!";
+            molErrors.push_back(ss.str());
+          }
+
+          config.addMolType(molID, molToSiteId, molToSitePos, momentOfInertia);
+        }
+
+        for_each(molErrors.begin(), molErrors.end(), pushMolError);
```

Comment:
this is `std::for_each`, right? Why can you use it without `std::` here? I don't see a `using` declaration??

Reply:
At the time of writing I was just copying the object code. It appears this is a result of Argument-Dependent Lookup (https://en.cppreference.com/w/cpp/language/adl). 

TLDR: `molErrors` is a `std::vector` i.e. in `std` namespace, so if no `for_each` is found, it looks for one within `std`. 

I'm personally easy as to if we should add the `std::` or not - up to you. Advice online is pretty sparse, but what exists is a pretty lukewarm opinion about relying on ADL